en/services/opendroneid.md

```diff
@@ -49,24 +50,46 @@ Self-ID | Self-ID | [OPEN_DRONE_ID_SELF_ID](../messages/common.md#OPEN_DRONE_ID_
   Slight compression is applied.
   Example code for this conversion can be found in the project: [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c).
 
 The [Open Drone ID Core C Library](https://github.com/opendroneid/opendroneid-core-c) contains code for decoding the MAVLink messages and "compressing" the data into data structures for transmission over Bluetooth or Wi-Fi (or vice-versa).
 
-The standards requires that the Location message is broadcast/published at least once per second.
-The rest of the messages must be broadcast/published once per 3 seconds (requirements from local legislation might be different.
-Please see [here](https://github.com/opendroneid/opendroneid-core-c#comparison)).
-Not all message types are mandatory to broadcast.
-
 The standards do not impose any requirements for a drone to be capable of receiving drone ID messages, nor any requirements for reacting to their content (requirements from local legislation might be different).
 
-An example Android receiver implementation for broadcast drone ID messages is available here: [OpenDroneID Android receiver application](https://github.com/opendroneid/receiver-android).
+An example Android receiver implementation for broadcast drone ID messages is [available here](https://play.google.com/store/apps/details?id=org.opendroneid.android_osm) ([source codes](https://github.com/opendroneid/receiver-android)).
 
 Code related to (Internet) Network Remote ID can be found in the [InterUSS Project](https://github.com/interuss) and https://github.com/uastech/standards (Unofficial reference for UAS-related APIs).
 
+## Message update rates {#update_rates}
+
+The standards require that the Location message is broadcast/published from the transmitter at least once per second.
+The rest of the messages must be broadcast/published once per 3 seconds (the rules in some regions have tightened this requirement to 1 second for also the Basic ID and the System message.
+Please see [here](https://github.com/opendroneid/opendroneid-core-c#comparison)).
+
+Not all message types or fields within a message type are mandatory to broadcast/publish.
+The mandatory message set vary from region to region.
+Please see the [summary](https://github.com/opendroneid/opendroneid-core-c#comparison).
+
+The standards require that the data contained in any transmitted/published message (on the air/internet) is not older than 1 second (e.g. location data, timestamps etc.).
+
+The integrator of the UAS must consider the combination of the above requirements when deciding the internal update rates for each component in the MAVLink network that generate data for the drone ID transmitter.
+It should be expected that there will be some delays from e.g. the GCS sending a System message via the flight controller to the transmitter component, which gather this and data from other messages together and generate the final message data set, which is transmitted/published on the air.
```

Comment:
I think it's pretty clear the GCS needs to send OPEN_DRONE_ID_SYSTEM faster than 1 Hz in order to meet the "not older than 1 second requirement", is this saying the integrator is required to have their own custom build of Mission Planner/QGC etc?

Reply:
It is not totally clear to me, on how abstract a level to keep this description. I don't really want to start going into details of specific control stacks (ArduPilot, PX4). I guess both teams will now work on a full-scale integration of drone ID and set suitable values for the various components that are being tested. However, anyone is free to replace any of these components with other variants and that person then ends up being the system integrator. 

I will try to think of a way to clarify that a bit.

Reply:
From a mavlink perspective there is an assumption that streaming rates are guidelines, and that often these aren't really thought about in detail because we assume that people will be doing some level of manual tuning in order to not swamp their channels. 

However if there is a hard requirement for certification then there should be a note to this effect on particular messages - not necessarily of the "number" but of the existence of the number.

You should avoid details of the implementation if at all possible. What should be the same is the "API" - so all information in the messages should be the same, and if messages cause something to happen, that behaviour should be the same.

Hope that makes sense.